src/compiler/watch.ts

```diff
@@ -174,6 +195,34 @@ namespace ts {
         return ExitStatus.Success;
     }
 
+    function summarizeDiagnosticsAcrossFiles(diagnostics: Diagnostic[], reporter: WatchStatusReporter, newLine: string, compilerOptions: CompilerOptions): void {
+        if (diagnostics.length === 1) {

```

Comment:
This is incorrect too.. This could be just compiler diagnostics or something.. Need not be in 1 file.

Reply:
I'm tempted to say simplify and don't even bother with pluralization based on count. Other languages don't necessarily pluralize the same way.

Reply:
> could just be compiler diagnostics

How should it count the numbers of errors from diagnostics? diagnostic.category === DiagnosticCategory.Error, ignoring warnings & below?

> simplify and don't even bother with pluralization

If a language doesn't pluralize for counts, then they would just have three localized messages with the same text. Is that bad? Seems fine to allow redundancy for nicer phrasing in most languages?

Reply:
I think the multiple diagnostic messages are the correct approach here. the other one is to use `error(s)` instead, but i would rather we add the extra line and make it look pretty.

As for knowing which is which, you can filter the diagnostics on `d.category === DiagnosticCategory.Error`.

One thing we can simplify is removing the number of files altogether. since diagnostics can be global and not attached to any file. this will reduce the number of message also by one, and will simplify this function.